core/frontend/src/components/parameter-editor/ParameterLoader.vue

```diff
@@ -0,0 +1,321 @@
+<template>
+  <v-dialog
+    v-model="should_open"
+    width="fit-content"
+    max-width="80%"
+    @click:outside="done"
+  >
+    <v-card>
+      <v-card-title class="pt-6">
+        Loading Parameters
+      </v-card-title>
+      <v-card-text v-if="Object.keys(different_param_set).length !== 0">
+        <v-row class="virtual-table-row">
+          <v-col class="virtual-table-cell checkbox-cell">
+            <v-checkbox
+              ref="selectAllCheckbox"
+              v-model="select_all"
+              class="virtual-table-cell"
+              label="Write"
+              :disabled="writing"
+              :indeterminate="select_all === null"
+              @click="toggleSelectAll"
+            />
+          </v-col>
+          <v-col class="virtual-table-cell name-cell">
+            <strong>Name</strong>
+          </v-col>
+          <v-col class="virtual-table-cell">
+            <strong>Value</strong>
+          </v-col>
+          <v-col class="virtual-table-cell">
+            <strong>New Value</strong>
+          </v-col>
+        </v-row>
+        <!-- display all parameters in a concise table using virtual scroller -->
+        <v-virtual-scroll
+          :items="printable(different_param_set)"
+          height="300"
+          item-height="30"
+          class="virtual-table"
+        >
+          <template #default="{ item }">
+            <v-row class="virtual-table-row">
+              <v-col class="virtual-table-cell">
+                <v-checkbox
+                  v-model="param_checkboxes[item.name]"
+                  class="checkbox-label checkbox-cell"
+                  :disabled="writing"
+                />
+              </v-col>
+              <v-col class="virtual-table-cell name-cell">
+                {{ item.name }}
+              </v-col>
+              <v-col class="virtual-table-cell">
+                {{ printParam(item.current_value) }}
+              </v-col>
+              <v-col class="virtual-table-cell">
+                {{ printParam(item.value) }}
+              </v-col>
+            </v-row>
+          </template>
+        </v-virtual-scroll>
+        <v-progress-linear
+          v-if="user_selected_params_length > 0 && initial_size > 0"
+          slot="progress"
+          :color="error ? 'red' : 'blue'"
+          height="20"
+          :value="progress"
+        >
+          <template #default>
+            <strong
+              class="ml-5"
+            >
+              {{ user_selected_params_length }} to go
+            </strong>
+          </template>
+        </v-progress-linear>
+      </v-card-text>
+      <v-card-text v-else-if="write_finished">
+        <v-alert
+          type="success"
+        >
+          Parameters written successfully
+        </v-alert>
```

Comment:
This triggers whenever there's no diff in the parameters, meaning that it shows "parameters written successfully" if you've executed the upload once in the past:

https://github.com/bluerobotics/BlueOS-docker/assets/5920286/63ff1dc4-d98d-4d57-b9ca-735af2ebd03c

Reply:
I added a check for initial_size > 0 to check if it actually attempted to write anything